### PR TITLE
refactor(conventional_commits): make schema_pattern more readable

### PR DESCRIPTION
**Pull request by bearomorphism**
## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [x] Add test cases to all the changes you introduce
- [x] Run `poetry all` locally to ensure this change passes linter check and tests
- [x] Manually test the changes:
  - [x] Verify the feature/bug fix works as expected in real-world scenarios
  - [x] Test edge cases and error conditions
  - [x] Ensure backward compatibility is maintained
  - [x] Document any manual testing steps performed
- [x] Update the documentation for the changes

## Steps to Test This Pull Request

Run `cz check --message "blahblahblah"` multiple times to check if it's working as expected. We don't use capture groups of `re.compile(self.schema_pattern)` so it's fine to adjust the parentheses.

### CODE REVIEW

File: commitizen/config/base_config.py [hunk 1]
```diff
@@ -19,6 +19,15 @@ def settings(self) -> Settings:
     def path(self) -> Path | None:
         return self._path
 
+    @path.setter
+    def path(self, path: str | Path) -> None:
+        """
+        mypy does not like this until 1.16
+        See https://github.com/python/mypy/pull/18510
+        TODO: remove "type: ignore" from the call sites when 1.16 is available
+        """
```
Comment (Lee-W) [inline commitizen/config/base_config.py]: I think we have already upgraded to 1.16
```suggestion
```

File: commitizen/config/json_config.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class JsonConfig(BaseConfig):
     def __init__(self, *, data: bytes | str, path: Path | str):
         super().__init__()
         self.is_empty_config = False
-        self.add_path(path)
+        self.path = path  # type: ignore
```
Comment (Lee-W) [inline commitizen/config/json_config.py]: Do we still need the type ignore here?

File: commitizen/config/toml_config.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class TomlConfig(BaseConfig):
     def __init__(self, *, data: bytes | str, path: Path | str):
         super().__init__()
         self.is_empty_config = False
-        self.add_path(path)
+        self.path = path  # type: ignore
```
Comment (Lee-W) [inline commitizen/config/toml_config.py]: same

File: commitizen/config/yaml_config.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ class YAMLConfig(BaseConfig):
     def __init__(self, *, data: bytes | str, path: Path | str):
         super().__init__()
         self.is_empty_config = False
-        self.add_path(path)
+        self.path = path  # type: ignore
```
Comment (Lee-W) [inline commitizen/config/yaml_config.py]: same

**Comment by Lee-W** [comment]
Oh... no wonder there are so many codes I think I have read. would need you help to resolve the conflict

**Review by Lee-W — APPROVED** [review]
